Lib/subprocess.py

```diff
@@ -809,6 +822,37 @@ def __init__(self, args, bufsize=-1, executable=None,
             else:
                 line_buffering = False
 
+        gids = []
+        if groups is not None:
+            if isinstance(groups, str):
+                raise ValueError("Groups must be a list, not a string")
+
+            for group in groups:
+                if isinstance(group, str):
+                    gids.append(grp.getgrnam(group).gr_gid)
+                elif isinstance(group, int):
+                    gids.append(group)
+                else:
+                    raise TypeError("Group must be a string or integer")
+
```

Comment:
Wouldn't it make sense to explicitly disallow empty `groups` by throwing a `ValueError`? A process always has a gid, so allowing empty groups might be confusing.

Reply:
Makes sense, this is done below.